lldb/source/Target/Target.cpp:844: function only ever returns true ?

Static analyser cppcheck says:

lldb/source/Target/Target.cpp:844:42: warning: Identical condition '!num_supported_hardware_watchpoints', second condition is always false [identicalConditionAfterEarlyExit]
  if (num_supported_hardware_watchpoints == 0) {
                                         ^

Source code is

 // If unable to determine the # of watchpoints available,
  // assume they are supported.
  if (!num_supported_hardware_watchpoints)
    return true;

  if (num_supported_hardware_watchpoints == 0) {
    error.SetErrorStringWithFormat(
        "Target supports (%u) hardware watchpoint slots.\n",
        *num_supported_hardware_watchpoints);
    return false;
  }
  return true;

The second if can never be true, so the function only ever returns true.
Suggest code rework.

--- a/lldb/source/Target/Target.cpp
+++ b/lldb/source/Target/Target.cpp
@@ -841,12 +841,14 @@ static bool CheckIfWatchpointsSupported(Target *target, Status &error) {
   if (!num_supported_hardware_watchpoints)
     return true;
 
-  if (num_supported_hardware_watchpoints == 0) {
-    error.SetErrorStringWithFormat(
-        "Target supports (%u) hardware watchpoint slots.\n",
-        *num_supported_hardware_watchpoints);
-    return false;
-  }
+  // If num_supported_hardware_watchpoints is zero, set an 
+  //error message and return false.
+
+  error.SetErrorStringWithFormat(
+      "Target supports (%u) hardware watchpoint slots.\n",
+      *num_supported_hardware_watchpoints);
+  return false;
+  
   return true;
 }
 

--- a/lldb/source/Target/Target.cpp
+++ b/lldb/source/Target/Target.cpp
@@ -841,14 +841,12 @@ static bool CheckIfWatchpointsSupported(Target *target, Status &error) {
   if (!num_supported_hardware_watchpoints)
     return true;
 
-  // If num_supported_hardware_watchpoints is zero, set an 
-  //error message and return false.
-
-  error.SetErrorStringWithFormat(
-      "Target supports (%u) hardware watchpoint slots.\n",
-      *num_supported_hardware_watchpoints);
-  return false;
-  
+  if (*num_supported_hardware_watchpoints == 0) {
+    error.SetErrorStringWithFormat(
+        "Target supports (%u) hardware watchpoint slots.\n",
+        *num_supported_hardware_watchpoints);
+    return false;
+  }
   return true;
 }
 

